src/main/java/fact/utils/SamplePedestalEvent.java

```diff
@@ -0,0 +1,385 @@
+                }
+
+                previousEventInfo.addNewInfo(startCellArray, stopCellArray, eventTime);
+            }
+        } catch (Exception e) {
+            log.error("Couldn't read file: "+fullpath);
+            throw new RuntimeException(e);
+        }
+
+        if (item==null) {
+            log.error("Couldn't read the pedestal item, nr: '"+noiseNr+"' file: "+fullpath);
+            throw new IOException("Couldn't read the pedestal item");
+        }
+        item.put("noise_db_event_id", noiseNr);
+        item.put("drspath", new File(drsPath));
+        item.put("prevEvents", previousEventInfo);
+        return item;
+    }
+
+    /**
+     * load the database and apply the condition and binning
+     */
+    private void prepareNoiseDatabase() {
+        // create an initial database, the initial capacity is set to a high value to reduce the amount of inital copying due to more elemets being added
+        this.database = new ArrayList<Data>(100000);
+        this.bins_index = new ArrayList<List<Integer>>(this.bins.length);
+        for(int i=0; i<this.bins.length; i++) {
+            this.bins_index.add(new ArrayList<Integer>(100));
+        }
+        try {
+	        log.info("Noise db: "+noiseDatabase.toString());
+            JSONStream databaseStream = new JSONStream(noiseDatabase);
+            databaseStream.init();
+
+            Data item = null;
+            while ((item = databaseStream.readNext()) != null) {
+                if (this.condition!=null) {
+                    Boolean b = (Boolean) this.condition.get(item);
+                    if (b == null ? false : !b.booleanValue()) // if condition doesn't applies ignore element
+                        continue;
+                }
+                // add the entry into the database
+                this.database.add(item);
+                // add index to the correct bin_index
+                double value = (double)item.get(this.dbBinningKey);
+                int binNum = this.getBin(value);
+                this.bins_index.get(binNum).add(this.database.size()-1);
+            }
+            log.info("Database consists of: "+this.database.size() + " Pedestals.");
+            for(int i=0;i<this.bins.length;i++){
+                log.info("Bin: "+i+", edge: " + this.bins[i] + " Size: " + this.bins_index.get(i).size());
+            }
+
+        } catch (Exception e) {
+            log.error("Got exception");
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Given a value data, calculate the bin it would end up in.
+     *
+     * @param data
+     * @return
+     */
+
+    protected int getBin(double data) {
+        for(int i=0; i<this.bins.length; i++) {
+            if (data<this.bins[i])
+                return i;
+        }
+        throw new RuntimeException("Data has a value outside of the binning");
+    }
+
+    public void setNoiseFolder(String dataFolder) {
```

Comment:
Why are getters and setters needed?

Reply:
Historically still in there, but are now gone.

Reply:
No, they are still here